static/extensions/jhub7682/MIDIdevices.js

```diff
@@ -0,0 +1,182 @@
+// Name: Midi Devices
+// ID: mididevicesjhub7682
+// Description: An extention that exposes MIDI devices
+// By: Jhub7682
+// License: MIT AND LGPL-3.0
+
+// Version V.1
+
+let lastNote = 0;
+let lastVelocity = 0;
+let pressedNotes = new Set();
+let pitchBendValue = 0;
+let controlValues = new Map();
+let programNumber = 0;
+let aftertouchValue = 0;
+let polyAftertouchValues = new Map();
+let midiDeviceName = "None";
+
+navigator.requestMIDIAccess()
+  .then((midiAccess) => {
+    for (let input of midiAccess.inputs.values()) {
+      midiDeviceName = input.name;
+
+      input.onmidimessage = (msg) => {
+        const [status, data1, data2] = msg.data;
+        const command = status & 0xF0;
+
+        switch (command) {
+          case 0x90: // Note On
+            if (data2 > 0) {
+              lastNote = data1;
+              lastVelocity = data2;
+              pressedNotes.add(data1);
+            } else {
+              pressedNotes.delete(data1); // Note Off via velocity 0
+            }
+            break;
+
+          case 0x80: // Note Off
+            pressedNotes.delete(data1);
+            break;
+
+          case 0xB0: // Control Change
+            controlValues.set(data1, data2);
+            break;
+
+          case 0xC0: // Program Change
+            programNumber = data1;
+            break;
+
+          case 0xD0: // Channel Pressure (Aftertouch)
+            aftertouchValue = data1;
+            break;
+
+          case 0xE0: // Pitch Bend
+            pitchBendValue = ((data2 << 7) | data1) - 8192;
+            break;
+
+          case 0xA0: // Polyphonic Aftertouch
+            polyAftertouchValues.set(data1, data2);
+            break;
+        }
+      };
+    }
+
+    Scratch.extensions.register({
+      getInfo() {
+        return {
+          id: "MDjhub7682",
```

Comment:
i would make this ID more clear with something like `jhub7682MidiDevices`
just so if someone gets an error with just the extension ID it makes more sense what the extension is actually called